ge/org.osate.ge/src/org/osate/ge/internal/ui/properties/ClassifierPrototypeBindingsPropertySection.java

```diff
@@ -0,0 +1,102 @@
+ * files ("Third Party Software"). The Third Party Software that is used by this program is dependent upon your system
+ * configuration. By using this program, You agree to comply with any and all relevant Third Party Software terms and
+ * conditions contained in any such Third Party Software or separate license file distributed with such Third Party
+ * Software. The parties who own the Third Party Software ("Third Party Licensors") are intended third party benefici-
+ * aries to this license with respect to the terms applicable to their Third Party Software. Third Party Software li-
+ * censes only apply to the Third Party Software and not any other portion of this program or this program as a whole.
+ */
+package org.osate.ge.internal.ui.properties;
+
+import org.eclipse.core.runtime.Adapters;
+import org.eclipse.jface.viewers.IFilter;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.CLabel;
+import org.eclipse.swt.layout.FormAttachment;
+import org.eclipse.swt.layout.FormData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.views.properties.tabbed.AbstractPropertySection;
+import org.eclipse.ui.views.properties.tabbed.ITabbedPropertyConstants;
+import org.eclipse.ui.views.properties.tabbed.TabbedPropertySheetPage;
+import org.osate.aadl2.Classifier;
+import org.osate.ge.BusinessObjectSelection;
+import org.osate.ge.internal.selection.AgeBusinessObjectSelection;
+import org.osate.ge.internal.viewModels.BusinessObjectSelectionPrototypeBindingsModel;
+import org.osate.ge.internal.viewModels.ClassifierPrototypeBindingsModel;
+import org.osate.ge.swt.classifiers.PrototypeBindingsField;
+import org.osate.ge.ui.properties.PropertySectionUtil;
+
+/**
+ * Property section for editing prototypes for classifiers.
+ *
+ */
+public class ClassifierPrototypeBindingsPropertySection extends AbstractPropertySection {
+	public static class Filter implements IFilter {
+		@Override
+		public boolean select(final Object toTest) {
+			return PropertySectionUtil.isBoCompatible(toTest,
+					bo -> bo instanceof Classifier
+					&& BusinessObjectSelectionPrototypeBindingsModel.hasAvailableBindings((Classifier) bo));
+		}
+	}
+
+	private BusinessObjectSelection selectedBos;
+	private final BusinessObjectSelectionPrototypeBindingsModel model = new ClassifierPrototypeBindingsModel(
+			new AgeBusinessObjectSelection());
+
+	@Override
+	public void createControls(final Composite parent, final TabbedPropertySheetPage aTabbedPropertySheetPage) {
+		super.createControls(parent, aTabbedPropertySheetPage);
+
+		final Composite composite = getWidgetFactory().createFlatFormComposite(parent);
+		final CLabel label = getWidgetFactory().createCLabel(composite, "Prototype\nBindings:");
+		final Control field = new PrototypeBindingsField<>(composite, model, null);
```

Comment:
Because null represents the root node. The field will display the root.

Reply:
Okay, but why pass it in instead of just setting it to null in the constructor?  I don't see it being used in a way that passes in a value, unless it's for future uses?

Reply:
Consistency with other widgets for all the prototype binding stuff. To be explicit about which node it is intended to show.